processing/src/main/java/io/druid/query/Queries.java

```diff
@@ -55,6 +61,18 @@ public static void verifyAggregations(
             "Missing fields [%s] for postAggregator [%s]", missing, postAgg.getName()
         );
         Preconditions.checkArgument(combinedAggNames.add(postAgg.getName()), "[%s] already defined", postAgg.getName());
+
+        if (postAgg instanceof HasDependentAggFactories) {
```

Comment:
@nishantmonu51 I made `ArithmeticPostAggregator` implements `HasDependentAggFactories`. It seems work now. But, do you think things are getting ugly?